### ISSUE TITLE
Calling SP with wrong parameters causes the server to crash

### ISSUE THREAD

**Issue by snotmare** [markdirish/node-odbc#58]
I have a stored procedure that requires 9 parameters. I passed in 8. The call hung for about 1 second and crashed...
[nodemon] app crashed - waiting for file changes before starting...

Can this return an error instead?

**Comment by markdirish** [comment]
Hi @snotmare ,

It _should_ return an error as written currently.

https://github.com/markdirish/node-odbc/blob/ca5c37727522ea309f90327d59b0eb2938da8969/src/odbc_connection.cpp#L998

On this line, it checks the number of parameters passed in against the number of parameters that the procedure takes (retrieved from `SQLProcedureColumns`).

Can you let me know the procedure signature (what are the datatypes and in/out types for each field), and what you passed in?

Thanks!

**Comment by snotmare** [comment]
Sure! This is the definition of the SP I'm trying to call...
```
CREATE PROCEDURE DUNC/GPR5016SP ( 
	INOUT GACT CHAR(2) , 
	IN IGNO CHAR(1) , 
	IN CUSR CHAR(10) , 
	IN NAME CHAR(10) , 
	IN LIBRARY CHAR(10) , 
	IN "TYPE" CHAR(10) , 
	IN CHARVALUE VARCHAR(2000) , 
	IN DECIMALVALUE DECIMAL(31, 9) , 
	IN INCREMENTVALUE DECIMAL(31, 9) ) 
	DYNAMIC RESULT SETS 1 
	LANGUAGE RPGLE 
	SPECIFIC DUNC/GPR5016SP 
	NOT DETERMINISTIC 
	MODIFIES SQL DATA 
	CALLED ON NULL INPUT 
	EXTERNAL NAME 'DUNC/GPR5016P' 
	PARAMETER STYLE GENERAL ; 
  
COMMENT ON SPECIFIC PROCEDURE DUNC/GPR5016SP 
	IS 'Dynamic Data Area Access' ; 
  
LABEL ON SPECIFIC PROCEDURE DUNC/GPR5016SP 
	IS 'Dynamic Data Area Access' ; 
  
GRANT EXECUTE   
ON SPECIFIC PROCEDURE DUNC/GPR5016SP 
TO PUBLIC ; 
  
GRANT ALTER , EXECUTE   
ON SPECIFIC PROCEDURE DUNC/GPR5016SP 
TO QPGMR WITH GRANT OPTION ;
```

This call works...
`let result = await this.connection.callProcedure(null, null, 'GPR5016SP', ['RE', 'Y', '', 'GPACMID', '', '', '', 0, 0]);`

This call does not (I removed the last parameter)...
`let result = await this.connection.callProcedure(null, null, 'GPR5016SP', ['RE', 'Y', '', 'GPACMID', '', '', '', 0]);`

I asked our system admin to do a trace to make sure we weren't missing any permissions or anything. We didn't see anything unusual, but here is some output from the trace in case it helps. I can see a call to SQLProcedureCols, plus some extra things because I'm testing on a server via a rest call that makes other SQL calls...
```
"Start Time"	"End Time"	"Runtime"	"Statement Outcome"	"SQLSTATE"	"SQLCODE"	"Operation"	"Statement Text"	"Variable Values"	"Full Opens"	"Pseudo Opens"	"Result Rows"	"ODP Implementation"	"Optimizer Information"	"Expression Evaluator"	"Scalar UDFs Not Inlined"	"Parse Required"	"Data Conversion"	"Access Plan Rebuild Reason"	"Rebuild Subcode"	"Access Plan Save Reason"	"Access Plan Length"	"Access Plan Rebuilt Length"	"SQL Statement Compression Maximum"	"QDT Rebuild"	"SQL Fast Delete Row Count"	"SQL Fast Delete Reason Code"	"Prepared Statement Not Reused Reason Code"	"Pseudo Open"	"Pseudo Close"	"Open Options"	"Row and Column Access Control"	"Isolation Level"	"Concurrent Access Requested"	"Concurrent Access Used"	"First N Rows Only"	"Optimize For N Rows"	"Cursor Type"	"Allow Copy Data"	"Blocking Enabled"	"Binding Type"	"Delay Prepare"	"Extended Indicators"	"Naming"	"User Profile"	"Dynamic User Profile"	"Default Collection"	"SQL Path"	"Current Temporal System Time"	"Date Format"	"Date Separator"	"Time Format"	"Time Separator"	"Decimal Point"	"Maximum Precision"	"Maximum Scale"	"Minimum Divide Scale"	"DECFLOAT Rounding Mode"	"DECFLOAT Warnings"	"Implicit XMLPARSE Option"	"Default XML Data CCSID"	"NLSS Table Schema"	"NLSS Table"	"Language ID"	"Country Or Region ID"	"Unicode Normalization"	"Close Cursor"	"Optimize LOBs"	"LOB Locator Threshold"	"Interface"	"Client Application Name"	"Client User ID"	"Client Workstation Name"	"Client Accounting"	"Client Program Name"	"Job"	"Job User"	"Job Number"	"User Name"	"CURRENT_USER"	"Thread ID"	"Activation Group"	"Program Schema"	"Program"	"Pool Size"	"Pool ID"	"IP Address"	"IP Port Number"	"Relational Database Name"	"User Defined Field"	"Cursor"	"Result Set Cursor Name"	"Statement Name"	"Statement Number"	"Statement Type"	"SQL Statement Originator"	"Hard Close Reason"	"Cursor Hard Close Subcode"	"Monitor Threshold"	"Transaction Identifier (XID)"	"NTS Lock Space ID"	"Refresh Count"	"Unique Count"	"Join Field"
"2019-12-24 08:35:51.665513"	"2019-12-24 08:35:51.706524"	0.041008	"Successful"	"0100C"	466	"CALL"	"CALL SYSIBM.SQLProcedureCols(NULL,?,?,?,?)"	"'%', 'GPR5016SP', '%', NULL"	""	""	0	"Nonreusable"	""	""	0	"No"	""	"NA"	"0000"	""	0	0	2	""	0	""	""	""	""	""	""	"No Commit"	""	""	0	0	""	"*OPTIMIZE"	"*ALLREAD"	"By row"	"Yes"	""	"System"	"N"	"U"	""	"QLPAR     ,SECURE    ,SYSTEM    ,QSYS      ,QSYS2     ,QUSRSYS   ,QHLPSYS   ,QIWS      ,QGPL      ,QTEMP     ,QGPL      ,DUNDDW    ,DUNCDW    ,DUNOPNSRC ,DUND      ,DUNC      ,APLUS01FDA,APLUS01M  ,APLUS01C  ,APLUS01   ,AVPTAX    ,AVPZIP    ,DBU11     ,TAATOOL"	""	"ISO"	"-"	"HMS"	":"	"."	31	31	0	"ROUND_HALF_EVEN"	"No"	"STRIP WHITESPACE"	1208	""	"*HEX"	"ENU"	""	"No"	""	"Yes"	0	""	"NODE"	""	""	""	""	"QZDASOINIT"	"QUSER"	"785132"	"user"	""	28624	2	"QSYS"	"QSQSIPROCC"	0	0	"10.1.220.8"	8471	"LEDEV01"	""	""	""	"STMT0002"	2	"Dynamic"	"User"	""	""	0	""	""	0	0	"D3C5C4C5E5F0F140D8E9C4C1E2D6C9D5C9E3D8E4E2C5D94040404040F7F8F5F1F3F2000000000000000100000000"
"2019-12-24 08:35:51.622864"	"2019-12-24 08:35:51.635886"	0.013016	"Successful"	"00000"	0	"PREPARE...DESCRIBE"	"SELECT CURRENT_SERVER, SPECIFIC_SCHEMA, SPECIFIC_NAME, IN_PARMS, OUT_PARMS, RESULT_SETS, VARCHAR(REMARKS, 30), ? FROM QSYS2/SYSPROCS WHERE SPECIFIC_SCHEMA IN (?,?,?,?,?,?,?,?,?,?,?,?,?,?,?) AND SPECIFIC_NAME = ? ORDER BY SPECIFIC_SCHEMA, SPECIFIC_NAME"	""	""	""	0	"Nonreusable"	""	""	0	"Yes"	""	"NA"	"0000"	""	0	0	2	""	0	""	"A0"	""	""	""	""	"No Commit"	""	""	0	0	""	"*OPTIMIZE"	"*ALLREAD"	"By row"	"Yes"	""	"System"	"N"	"U"	""	"QLPAR     ,SECURE    ,SYSTEM    ,QSYS      ,QSYS2     ,QUSRSYS   ,QHLPSYS   ,QIWS      ,QGPL      ,QTEMP     ,QGPL      ,DUNDDW    ,DUNCDW    ,DUNOPNSRC ,DUND      ,DUNC      ,APLUS01FDA,APLUS01M  ,APLUS01C  ,APLUS01   ,AVPTAX    ,AVPZIP    ,DBU11     ,TAATOOL"	""	"ISO"	"-"	"HMS"	":"	"."	31	31	0	"ROUND_HALF_EVEN"	"No"	"STRIP WHITESPACE"	1208	""	"*HEX"	"ENU"	""	"No"	""	"Yes"	0	""	"NODE"	""	""	""	""	"QZDASOINIT"	"QUSER"	"785132"	"user"	""	28624	2	"QSYS"	"QZDASRV"	0	0	"10.1.220.8"	8471	"LEDEV01"	""	""	""	"STMT0002"	1	"Dynamic"	"User"	""	""	0	""	""	0	0	"D3C5C4C5E5F0F140D8E9C4C1E2D6C9D5C9E3D8E4E2C5D94040404040F7F8F5F1F3F2000000000000000000000000"
"2019-12-24 08:35:51.559508"	"2019-12-24 08:35:51.567292"	0.007784	"Successful"	"00000"	0	"OPEN"	"select RCID, STATUS, USRRCID, JWTID, EXPTS, ENTS
from GPPJWTL where status = ? or expts <= current_timestamp"	"'I'"	1	""	0	"Nonreusable"	"SQE"	""	0	"No"	""	"NA"	"0000"	""	0	23296	2	""	0	""	""	"No"	"No"	"R"	""	"No Commit"	"WAIT FOR OUTCOME"	"Not Applicable"	0	0	"Non-Scrollable ASENSITIVE"	"*OPTIMIZE"	"*ALLREAD"	"By row"	"Yes"	""	"System"	"N"	"U"	""	"QLPAR     ,SECURE    ,SYSTEM    ,QSYS      ,QSYS2     ,QUSRSYS   ,QHLPSYS   ,QIWS      ,QGPL      ,QTEMP     ,QGPL      ,DUNOPNSRC ,WEBTEST   ,DUND      ,DUNC      ,APLUS01FDA,APLUS01M  ,APLUS01C  ,APLUS01   ,CALLAB    ,AVPTAX    ,SOFTSET   ,AVPZIP"	""	"ISO"	"/"	"USA"	"."	"."	31	31	0	"ROUND_HALF_EVEN"	"No"	"STRIP WHITESPACE"	1208	""	"*HEX"	"ENU"	""	"No"	""	"Yes"	0	""	""	""	""	""	""	"QZDASOINIT"	"QUSER"	"785118"	"user"	""	68	2	"QSYS"	"QZDASRV"	0	0	"10.1.220.8"	8471	"LEDEV01"	""	"CRSR0001"	""	"STMT0001"	3	"Dynamic"	"User"	""	""	0	""	""	0	7	"D3C5C4C5E5F0F140D8E9C4C1E2D6C9D5C9E3D8E4E2C5D94040404040F7F8F5F1F1F8000000070000000000000000"
"2019-12-24 08:35:51.643151"	"2019-12-24 08:35:51.649991"	0.006840	"Successful"	"00000"	0	"OPEN"	"SELECT CURRENT_SERVER, SPECIFIC_SCHEMA, SPECIFIC_NAME, IN_PARMS, OUT_PARMS, RESULT_SETS, VARCHAR(REMARKS, 30), ? FROM QSYS2/SYSPROCS WHERE SPECIFIC_SCHEMA IN (?,?,?,?,?,?,?,?,?,?,?,?,?,?,?) AND SPECIFIC_NAME = ? ORDER BY SPECIFIC_SCHEMA, SPECIFIC_NAME"	"1, 'QGPL', 'QTEMP', 'DUNDDW', 'DUNCDW', 'DUNOPNSRC', 'DUND', 'DUNC', 'APLUS01FDA', 'APLUS01M', 'APLUS01C', 'APLUS01', 'AVPTAX', 'AVPZIP', 'DBU11', 'TAATOOL', 'GPR5016SP'"	1	""	0	"Nonreusable"	"SQE"	""	0	"No"	""	"NA"	"0000"	""	0	66272	2	""	0	""	""	"No"	"No"	"R"	""	"No Commit"	"WAIT FOR OUTCOME"	"Not Applicable"	0	0	"Non-Scrollable ASENSITIVE"	"*OPTIMIZE"	"*ALLREAD"	"By row"	"Yes"	""	"System"	"N"	"U"	""	"QLPAR     ,SECURE    ,SYSTEM    ,QSYS      ,QSYS2     ,QUSRSYS   ,QHLPSYS   ,QIWS      ,QGPL      ,QTEMP     ,QGPL      ,DUNDDW    ,DUNCDW    ,DUNOPNSRC ,DUND      ,DUNC      ,APLUS01FDA,APLUS01M  ,APLUS01C  ,APLUS01   ,AVPTAX    ,AVPZIP    ,DBU11     ,TAATOOL"	""	"ISO"	"-"	"HMS"	":"	"."	31	31	0	"ROUND_HALF_EVEN"	"No"	"STRIP WHITESPACE"	1208	""	"*HEX"	"ENU"	""	"No"	""	"Yes"	0	""	"NODE"	""	""	""	""	"QZDASOINIT"	"QUSER"	"785132"	"user"	""	28624	2	"QSYS"	"QZDASRV"	0	0	"10.1.220.8"	8471	"LEDEV01"	""	"SQL_CURA25A8730"	""	"STMT0002"	1	"Dynamic"	"User"	""	""	0	""	""	0	2	"D3C5C4C5E5F0F140D8E9C4C1E2D6C9D5C9E3D8E4E2C5D94040404040F7F8F5F1F3F2000000020000000000000000"
"2019-12-24 08:35:51.545150"	"2019-12-24 08:35:51.548252"	0.003104	"Successful"	"00000"	0	"PREPARE...DESCRIBE"	"select RCID, STATUS, USRRCID, JWTID, EXPTS, ENTS
from GPPJWTL where status = ? or expts <= current_timestamp"	""	""	""	0	"Nonreusable"	""	""	0	"Yes"	""	"NA"	"0000"	""	0	0	2	""	0	""	""	""	""	""	""	"No Commit"	""	""	0	0	""	"*OPTIMIZE"	"*ALLREAD"	"By row"	"Yes"	""	"System"	"N"	"U"	""	"QLPAR     ,SECURE    ,SYSTEM    ,QSYS      ,QSYS2     ,QUSRSYS   ,QHLPSYS   ,QIWS      ,QGPL      ,QTEMP     ,QGPL      ,DUNOPNSRC ,WEBTEST   ,DUND      ,DUNC      ,APLUS01FDA,APLUS01M  ,APLUS01C  ,APLUS01   ,CALLAB    ,AVPTAX    ,SOFTSET   ,AVPZIP"	""	"ISO"	"/"	"USA"	"."	"."	31	31	0	"ROUND_HALF_EVEN"	"No"	"STRIP WHITESPACE"	1208	""	"*HEX"	"ENU"	""	"No"	""	"Yes"	0	"IBM Toolbox for Java:JDBC:07010009"	""	""	""	""	""	"QZDASOINIT"	"QUSER"	"785118"	"user"	""	68	2	"QSYS"	"QZDASRV"	0	0	"10.1.220.8"	8471	"LEDEV01"	""	""	""	"STMT0001"	3	"Dynamic"	"User"	""	""	0	""	""	0	0	"D3C5C4C5E5F0F140D8E9C4C1E2D6C9D5C9E3D8E4E2C5D94040404040F7F8F5F1F1F8000000000000000000000000"
"2019-12-24 08:35:51.658895"	"2019-12-24 08:35:51.660044"	0.001152	"Successful"	"00000"	0	"PREPARE...DESCRIBE"	"CALL SYSIBM.SQLProcedureCols(NULL,?,?,?,?)"	""	""	""	0	"Nonreusable"	""	""	0	"Yes"	""	"NA"	"0000"	""	0	0	2	""	0	""	"A0"	""	""	""	""	"No Commit"	""	""	0	0	""	"*OPTIMIZE"	"*ALLREAD"	"By row"	"Yes"	""	"System"	"N"	"U"	""	"QLPAR     ,SECURE    ,SYSTEM    ,QSYS      ,QSYS2     ,QUSRSYS   ,QHLPSYS   ,QIWS      ,QGPL      ,QTEMP     ,QGPL      ,DUNDDW    ,DUNCDW    ,DUNOPNSRC ,DUND      ,DUNC      ,APLUS01FDA,APLUS01M  ,APLUS01C  ,APLUS01   ,AVPTAX    ,AVPZIP    ,DBU11     ,TAATOOL"	""	"ISO"	"-"	"HMS"	":"	"."	31	31	0	"ROUND_HALF_EVEN"	"No"	"STRIP WHITESPACE"	1208	""	"*HEX"	"ENU"	""	"No"	""	"Yes"	0	""	"NODE"	""	""	""	""	"QZDASOINIT"	"QUSER"	"785132"	"user"	""	28624	2	"QSYS"	"QZDASRV"	0	0	"10.1.220.8"	8471	"LEDEV01"	""	""	""	"STMT0002"	2	"Dynamic"	"User"	""	""	0	""	""	0	0	"D3C5C4C5E5F0F140D8E9C4C1E2D6C9D5C9E3D8E4E2C5D94040404040F7F8F5F1F3F2000000000000000000000000"
"2019-12-24 08:35:51.570340"	"2019-12-24 08:35:51.570877"	0.000536	"Successful"	"00000"	0	"CLOSE"	"CLOSE CRSR0001"	""	""	""	0	"Nonreusable"	""	""	0	"No"	""	"NA"	"0000"	""	0	0	2	""	0	""	""	""	"Yes"	"R"	""	"No Commit"	""	""	0	0	""	"*OPTIMIZE"	"*ALLREAD"	"By row"	"Yes"	""	"System"	"N"	"U"	""	"QLPAR     ,SECURE    ,SYSTEM    ,QSYS      ,QSYS2     ,QUSRSYS   ,QHLPSYS   ,QIWS      ,QGPL      ,QTEMP     ,QGPL      ,DUNOPNSRC ,WEBTEST   ,DUND      ,DUNC      ,APLUS01FDA,APLUS01M  ,APLUS01C  ,APLUS01   ,CALLAB    ,AVPTAX    ,SOFTSET   ,AVPZIP"	""	"ISO"	"/"	"USA"	"."	"."	31	31	0	"ROUND_HALF_EVEN"	"No"	"STRIP WHITESPACE"	1208	""	"*HEX"	"ENU"	""	"No"	""	"Yes"	0	""	""	""	""	""	""	"QZDASOINIT"	"QUSER"	"785118"	"user"	""	68	2	"QSYS"	"QZDASRV"	0	0	"10.1.220.8"	8471	"LEDEV01"	""	"CRSR0001"	""	""	0	""	"User"	""	""	0	""	""	0	7	"D3C5C4C5E5F0F140D8E9C4C1E2D6C9D5C9E3D8E4E2C5D94040404040F7F8F5F1F1F8000000070000000000000000"
"2019-12-24 08:35:51.651510"	"2019-12-24 08:35:51.651888"	0.000384	"Successful"	"00000"	0	"CLOSE"	"CLOSE SQL_CURA25A8730"	""	""	""	0	"Nonreusable"	""	""	0	"No"	""	"NA"	"0000"	""	0	0	2	""	0	""	""	""	"No"	"R"	""	"No Commit"	""	""	0	0	""	"*OPTIMIZE"	"*ALLREAD"	"By row"	"Yes"	""	"System"	"N"	"U"	""	"QLPAR     ,SECURE    ,SYSTEM    ,QSYS      ,QSYS2     ,QUSRSYS   ,QHLPSYS   ,QIWS      ,QGPL      ,QTEMP     ,QGPL      ,DUNDDW    ,DUNCDW    ,DUNOPNSRC ,DUND      ,DUNC      ,APLUS01FDA,APLUS01M  ,APLUS01C  ,APLUS01   ,AVPTAX    ,AVPZIP    ,DBU11     ,TAATOOL"	""	"ISO"	"-"	"HMS"	":"	"."	31	31	0	"ROUND_HALF_EVEN"	"No"	"STRIP WHITESPACE"	1208	""	"*HEX"	"ENU"	""	"No"	""	"Yes"	0	""	"NODE"	""	""	""	""	"QZDASOINIT"	"QUSER"	"785132"	"user"	""	28624	2	"QSYS"	"QZDASRV"	0	0	"10.1.220.8"	8471	"LEDEV01"	""	"SQL_CURA25A8730"	""	""	0	""	"User"	"First full close"	""	0	""	""	0	2	"D3C5C4C5E5F0F140D8E9C4C1E2D6C9D5C9E3D8E4E2C5D94040404040F7F8F5F1F3F2000000020000000000000000"
"2019-12-24 08:35:51.570279"	"2019-12-24 08:35:51.570417"	0.000064	"Successful"	"00000"	0	"FETCH"	""	""	""	""	1	"Nonreusable"	""	""	0	"No"	""	"NA"	"0000"	""	0	0	2	""	0	""	""	""	""	"R"	""	"No Commit"	""	""	0	0	""	"*OPTIMIZE"	"*ALLREAD"	"By row"	"Yes"	""	"System"	"N"	"U"	""	"QLPAR     ,SECURE    ,SYSTEM    ,QSYS      ,QSYS2     ,QUSRSYS   ,QHLPSYS   ,QIWS      ,QGPL      ,QTEMP     ,QGPL      ,DUNOPNSRC ,WEBTEST   ,DUND      ,DUNC      ,APLUS01FDA,APLUS01M  ,APLUS01C  ,APLUS01   ,CALLAB    ,AVPTAX    ,SOFTSET   ,AVPZIP"	""	"ISO"	"/"	"USA"	"."	"."	31	31	0	"ROUND_HALF_EVEN"	"No"	"STRIP WHITESPACE"	1208	""	"*HEX"	"ENU"	""	"No"	""	"Yes"	0	""	""	""	""	""	""	"QZDASOINIT"	"QUSER"	"785118"	"user"	""	68	2	"QSYS"	"QZDASRV"	0	0	"10.1.220.8"	8471	"LEDEV01"	""	"CRSR0001"	""	"STMT0001"	0	""	"User"	""	""	0	""	""	0	7	"D3C5C4C5E5F0F140D8E9C4C1E2D6C9D5C9E3D8E4E2C5D94040404040F7F8F5F1F1F8000000070000000000000000"
"2019-12-24 08:35:51.651452"	"2019-12-24 08:35:51.651848"	0.000056	"Successful"	"00000"	0	"FETCH"	""	""	""	""	1	"Nonreusable"	""	""	0	"No"	""	"NA"	"0000"	""	0	0	2	""	0	""	""	""	""	"R"	""	"No Commit"	""	""	0	0	""	"*OPTIMIZE"	"*ALLREAD"	"By row"	"Yes"	""	"System"	"N"	"U"	""	"QLPAR     ,SECURE    ,SYSTEM    ,QSYS      ,QSYS2     ,QUSRSYS   ,QHLPSYS   ,QIWS      ,QGPL      ,QTEMP     ,QGPL      ,DUNDDW    ,DUNCDW    ,DUNOPNSRC ,DUND      ,DUNC      ,APLUS01FDA,APLUS01M  ,APLUS01C  ,APLUS01   ,AVPTAX    ,AVPZIP    ,DBU11     ,TAATOOL"	""	"ISO"	"-"	"HMS"	":"	"."	31	31	0	"ROUND_HALF_EVEN"	"No"	"STRIP WHITESPACE"	1208	""	"*HEX"	"ENU"	""	"No"	""	"Yes"	0	""	"NODE"	""	""	""	""	"QZDASOINIT"	"QUSER"	"785132"	"user"	""	28624	2	"QSYS"	"QZDASRV"	0	0	"10.1.220.8"	8471	"LEDEV01"	""	"SQL_CURA25A8730"	""	"STMT0002"	0	""	"User"	""	""	0	""	""	0	2	"D3C5C4C5E5F0F140D8E9C4C1E2D6C9D5C9E3D8E4E2C5D94040404040F7F8F5F1F3F2000000020000000000000000"
"2019-12-24 08:35:51.651516"	"2019-12-24 08:35:51.651516"	0.000000	"Successful"	"00000"	0	"CLOSE (Hard)"	"HARD CLOSE          1 CURSORS"	""	""	""	0	"Nonreusable"	""	""	0	"No"	""	"NA"	"0000"	""	0	0	2	""	0	""	""	""	""	""	""	""	""	""	0	0	""	""	"B"	"By row"	""	""	""	""	""	""	""	""	""	""	""	""	""	0	0	0	""	"No"	"N"	1208	""	"*HEX"	"ENU"	""	"No"	""	""	0	""	"NODE"	""	""	""	""	"QZDASOINIT"	"QUSER"	"785132"	"user"	""	28624	0	""	""	0	0	"10.1.220.8"	8471	"LEDEV01"	""	"SQL_CURA25A8730"	""	"STMT0002"	0	""	"User"	"First full close"	"SQCLSCSR : 39"	0	""	""	0	2	"D3C5C4C5E5F0F140D8E9C4C1E2D6C9D5C9E3D8E4E2C5D94040404040F7F8F5F1F3F2000000020000000000000000"
"2019-12-24 08:35:53.123920"	"2019-12-24 08:35:53.123920"	0.000000	"Successful"	"00000"	0	"CLOSE (Hard)"	"HARD CLOSE          1 CURSORS"	""	""	""	0	"Nonreusable"	""	""	0	"No"	""	"NA"	"0000"	""	0	0	2	""	0	""	""	""	""	""	""	""	""	""	0	0	""	""	"B"	"By row"	""	""	""	""	""	""	""	""	""	""	""	""	""	0	0	0	""	"No"	"N"	1208	""	"*HEX"	"ENU"	""	"No"	""	""	0	""	""	""	""	""	""	"QZDASOINIT"	"QUSER"	"785118"	"user"	""	68	0	""	""	0	0	""	0	"LEDEV01"	""	"CRSR0001"	""	"STMT0001"	0	""	"User"	"Exit processing"	"QSQXIT   : A3"	0	""	""	0	7	"D3C5C4C5E5F0F140D8E9C4C1E2D6C9D5C9E3D8E4E2C5D94040404040F7F8F5F1F1F8000000070000000000000000"
```

This may be another clue... I've made this incorrect SP call probably 20 times now and 1 of those times, I did see an error returned saying the parameters didn't match. The server still crashed after that. Maybe a timing thing?

**Comment by snotmare** [comment]
Good morning! I have installed the latest version (2.4.3) and confirmed this is still an issue. I also noticed that the server will crash if I try to call a stored procedure that doesn't exist. It appears most everything else is working, so I think this is getting closer to being production ready! I just can't have our servers crashing on us :/.

**Comment by markdirish** [comment]
I think I have solved this in this PR: https://github.com/markdirish/node-odbc/pull/265

The problem was an uninitialized pointer that would be freed if callProcedure failed by either passing in a wrong procedure name or incorrect number of parameters.

Will get it merged in and into the next version!

**Comment by markdirish** [comment]
Nice! Going to close this issue, as always feel free to reopen or create a new issue if you run into more issues!
